odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -1399,3 +1399,34 @@ class TriggerRight(models.Model):
     def _compute_left_size(self):
         for record in self:
             record.left_size = len(record.left_ids)
+
+class Crew(models.Model):
+    _name = 'test_new_api.crew'
+    _description = 'All yaaaaaarrrrr by ship'
+    _table = 'test_new_api_crew'
+
+    pirate_id = fields.Many2one('test_new_api.pirate')
+    prisonner_id = fields.Many2one('test_new_api.prisonner')
+    ship_id = fields.Many2one('test_new_api.ship')
+
+class Ship(models.Model):
+    _name = 'test_new_api.ship'
+    _description = 'Yaaaarrr machine'
+
+    name = fields.Char('Name')
+    pirate_ids = fields.Many2many('test_new_api.pirate', 'test_new_api_crew', 'ship_id', 'pirate_id')
+    prisonner_ids = fields.Many2many('test_new_api.prisonner', 'test_new_api_crew', 'ship_id', 'prisonner_id')
+
+class Pirate(models.Model):
+    _name = 'test_new_api.pirate'
+    _description = 'Yaaarrr'
+
+    name = fields.Char('Name')
+    ship_ids = fields.Many2many('test_new_api.ship', 'test_new_api_crew', 'pirate_id', 'ship_id')
```

Comment:
Note: It is kind of strange in this modelling that a pirate has multiple ships at once.

Reply:
Well, pirates did have multiple ships, Edward Teach actually started with 2 ships. Zheng Yi Sao commanded around 1600 ships at her power peak.

Reply:
Fair enough, I suppose it will be time to request review here.